runtime/common/src/xcm_configs.rs

```diff
@@ -205,3 +205,63 @@ impl<
 		OnUnbalanced::on_unbalanced(Currency::issue(self.1));
 	}
 }
+
+use frame_support::pallet_prelude::*;
+use sp_std::prelude::*;
+#[derive(Clone, Default, Eq, Debug, PartialEq, Ord, PartialOrd, Encode, Decode, TypeInfo)]
+pub struct AssetRegistrarMetadata {
+	pub name: Vec<u8>,
+	pub symbol: Vec<u8>,
+	pub decimals: u8,
+	pub is_frozen: bool,
+}
+
+// Our AssetType. For now we only handle Xcm Assets
+#[derive(Clone, Eq, Debug, PartialEq, Ord, PartialOrd, Encode, Decode, TypeInfo)]
+pub enum AssetType {
+	Xcm(MultiLocation),
+}
+impl Default for AssetType {
+	fn default() -> Self {
+		Self::Xcm(MultiLocation::here())
+	}
+}
+
+impl From<MultiLocation> for AssetType {
+	fn from(location: MultiLocation) -> Self {
+		Self::Xcm(location)
+	}
+}
+impl Into<Option<MultiLocation>> for AssetType {
+	fn into(self) -> Option<MultiLocation> {
+		match self {
+			Self::Xcm(location) => Some(location),
+		}
+	}
+}
+
+frame_support::parameter_types! {
+	pub UsdtLocation: MultiLocation = MultiLocation::new(
+		1,
+		X3(Parachain(1000), PalletInstance(50), GeneralIndex(1984))
+	);
+}
+
+use sp_runtime::traits::Hash;
+// Implementation on how to retrieve the AssetId from an AssetType
+// We simply hash the AssetType and take the lowest 128 bits
+impl From<AssetType> for crate::AssetId {
+	fn from(asset: AssetType) -> crate::AssetId {
+		match asset {
+			AssetType::Xcm(id) =>
+				if id == UsdtLocation::get() {
```

Comment:
How about this:

`if id == MultiLocation::new(1, X3(Parachain(1000), PalletInstance(50), GeneralIndex(1984)))`

Would you like to add some comment about what is the 1000, 50, 1984 comes from?